### ISSUE TITLE
[Bug]: after dn crash, cluster cannot be connected for a long time

### ISSUE THREAD

**Issue by dengn** [matrixorigin/matrixone#9794]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): nightly-54b5e8c
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

follow up #9793, 

after dn crashed and rebooting, the cluster doesn't come back to life.
<img width="1043" alt="image" src="https://github.com/matrixorigin/matrixone/assets/4965857/c1d667f8-5781-4028-9515-b61329f82128">

 the connection is stuck at:
<img width="863" alt="企业微信截图_8a4c6671-7cae-4f56-b037-c8847a2f7f11" src="https://github.com/matrixorigin/matrixone/assets/4965857/e9b08d0c-e9c6-4220-93b2-632f7f436981">

the log of dn, cn are below:
[dn.log.202306011615.txt](https://github.com/matrixorigin/matrixone/files/11622744/dn.log.202306011615.txt)
[cn.log.202306011615.txt](https://github.com/matrixorigin/matrixone/files/11622745/cn.log.202306011615.txt)

**Comment by gouhongshen** [comment]
from the log, could be the same problem as #8599

**Comment by gouhongshen** [comment]
Have reproduced it:
1. in the current main branch, for MySQL connection operation, restarting DN by deleting its metadata has the same effect as turning down the DN service.
2. using PR #9775, the connection can resume within HAKeepers DN timeout. 

waiting for the merge.

**Comment by gouhongshen** [comment]
waiting for the merge

**Comment by gouhongshen** [comment]
fixed by #9775

**Comment by gouhongshen** [comment]
previous PR have been reverted for problem investigation. re-PR-ing.

**Comment by gouhongshen** [comment]
fixed by #10105 

@dengn plz help to verify it

**Comment by dengn** [comment]
fixed. after dn crash, cluster can be connected. but it still took about 2 minutes.
